src/compiler/crystal/tools/doc/method.cr

```diff
@@ -175,4 +176,15 @@ class Crystal::Doc::Method
   def has_args?
     !@def.args.empty? || @def.block_arg || @def.yields
   end
+
+  def to_json(io)
+    {
+      name:     name,
+      args:     args_to_s,
```

Comment:
This should probably be better introspectable, I would imagine an array of `{name: "foo", type: "A|B", default: "nil", splat: false, double_splat: false, keyword: false}`, any attribute I missed?